### PR TITLE
[WIP] Add code coverage

### PR DESCRIPTION
**Pull request by mshitrit**
Using codacy to add code coverage to the repository, also each PR will have the delta of how it effects the previous code coverage.

### CODE REVIEW

File: .github/workflows/pre-submit.yaml [hunk 1]
```diff
@@ -21,7 +21,10 @@ jobs:
         go-version: 1.15
 
     - name: Build and test
-      run: make manifests generate docker-build
+      run: make manifests generate docker-build test
```
Comment (n1r1) [inline .github/workflows/pre-submit.yaml:24]: `docker-build` already runs `test`, is this change required?

Comment (mshitrit) [reply .github/workflows/pre-submit.yaml:24]: At the moment Codacy has issues with their code coverage plugin so this PR is nor relevant (that's why I've marked it WIP).
I'm giving them some grace time to fix it.
I suggest we'll discuss the PR content in case it'll be relevant.

**Comment by n1r1** [comment]
@mshitrit what's the status of this?
can we close it?

**Comment by mshitrit** [comment]
> @mshitrit what's the status of this? can we close it?

Sure.
Done
